### PR TITLE
added support for global proxy only subnets

### PR DESCRIPTION
**Pull request by dgulli**
To support the new cross-regional load balancers, i need to be able to create global managed proxy subnet types, rather than regional, as per the doco 

[here](https://cloud.google.com/load-balancing/docs/l7-internal/setting-up-l7-cross-reg-internal#configure-a-network) and 
[here](https://cloud.google.com/load-balancing/docs/proxy-only-subnets#envoy-lb)

---
**Checklist**

I applicable, I acknowledge that I have:
- [X] Read the [contributing guide](https://github.com/GoogleCloudPlatform/cloud-foundation-fabric/blob/master/CONTRIBUTING.md)
- [X] Ran `terraform fmt` on all modified files
- [X] Regenerated the relevant README.md files using [`tools/tfdoc.py`](https://github.com/GoogleCloudPlatform/cloud-foundation-fabric/blob/master/CONTRIBUTING.md#fabric-tools)
- [X] Made sure all relevant tests pass

### CODE REVIEW

File: modules/net-vpc/subnets.tf
Comment (wiktorn) [inline modules/net-vpc/subnets.tf:83]: I guess you wanted to introduce new variable here, instead reuse `var.subnets_psc`?

**Comment by ludoo** [comment]
I think we should internalize the subnet type in the subnet structure instead of having a ton of variables, which was the legacy way of doing this before `optional()`. This is ok for now, just keep in mind we will probably change the whole interface soonish.

File: modules/net-vpc/subnets.tf
Comment (alloveras) [reply modules/net-vpc/subnets.tf:83]: Also, couldn't we re-use `var.subnets_proxy_only` and have the filtering predicate have something like `contains(v.purpose, ["REGIONAL_MANAGED_PROXY", "GLOBAL_MANAGED_PROXY"])`?

File: modules/net-vpc/variables.tf
Comment (juliocc) [inline modules/net-vpc/variables.tf:230]: Do we really need a new variable for this? I'd rather reuse the existing `subnets_proxy_only` and add a new `global` attribute with type `optional(boolean, FALSE)`.

**Review by juliocc — CHANGES_REQUESTED** [review]
Left a comment below. Also, I'd like to see at least one test exercising the new type of subnet

File: modules/net-vpc/variables.tf
Comment (dgulli) [reply modules/net-vpc/variables.tf:230]: That's fair - I'll have a read how to implement optional into the existing proxy subnet.

Comment (dgulli) [reply modules/net-vpc/variables.tf]: @juliocc  - i've refactored to use the existing proxy_only subnet, and removed all the other work done to reflect. Introduced test to create the additional subnet and regenerated the doco

File: modules/net-vpc/subnets.tf
Comment (wiktorn) [inline modules/net-vpc/subnets.tf:136]: This can be rewritten as:
`each.value.global ? "GLOBAL_MANAGED_PROXY" : "REGIONAL_MANAGED_PROXY"`

which is shorter

File: modules/net-vpc/subnets.tf
Comment (wiktorn) [inline modules/net-vpc/subnets.tf:36]: This part refers to YAML file parsing, for backward compatibility please keep `v.purpose` here.

As you are adding `purpose: GLOBAL_MANAGED_PROXY` you need to add specific handling below, when creating `subnets_proxy_only` from `local._factory_subnets`. You need to add `active` true/false depending on type of what is the purpose (global or regional proxy)

**Review by wiktorn — CHANGES_REQUESTED** [review]

File: modules/net-vpc/subnets.tf
Comment (dgulli) [reply modules/net-vpc/subnets.tf:136]: thanks for that - i find that the way i've put this is a little easier for read for peeps without deep TF experience!

File: modules/net-vpc/subnets.tf
Comment (ludoo) [inline modules/net-vpc/subnets.tf:142]: nit: can you add a line break between `"REGIONAL_MANAGED_PROXY"` and the following `)` then run `terraform ftm`?

**Review by ludoo — APPROVED** [review]

File: modules/net-vpc/subnets.tf
Comment (dgulli) [reply modules/net-vpc/subnets.tf]: done!

**Comment by ludoo** [comment]
hit the merge button once all is green, thanks for this!

**Comment by ludoo** [comment]
nit:

```hcl
each.value.global != false
    ? "GLOBAL_MANAGED_PROXY"
    : "REGIONAL_MANAGED_PROXY"
```

will default to global if `each.value.global` is `null`, which is the opposite of the default in the variable type. I would remove the `!= false` or change it to `== true`. We can do this after merging.

**Comment by dgulli** [comment]
> nit:
> 
> ```terraform
> each.value.global != false
>     ? "GLOBAL_MANAGED_PROXY"
>     : "REGIONAL_MANAGED_PROXY"
> ```
> 
> will default to global if `each.value.global` is `null`, which is the opposite of the default in the variable type. I would remove the `!= false` or change it to `== true`. We can do this after merging.

done - https://github.com/GoogleCloudPlatform/cloud-foundation-fabric/pull/1676

**Pull request merged**
